src/SFML/Graphics/Shape.cpp

```diff
@@ -143,6 +142,49 @@ FloatRect Shape::getGlobalBounds() const
     return getTransform().transformRect(getLocalBounds());
 }
 
+Vector2f Shape::getGeometricCenter() const
+{
+    const auto count = getPointCount();
+
+    switch (count)
+    {
+        case 0:
+            assert(false);
+            return Vector2f{};
+        case 1:
+            return getPoint(0);
+        case 2:
+            return (getPoint(0) + getPoint(1)) / 2.f;
+        default: // more than two points
+            Vector2f centroid;
+            float    twiceArea = 0;
+
+            auto previous_point = getPoint(count - 1);
+            auto minPoint       = previous_point;
+            auto maxPoint       = previous_point;
+            for (std::size_t i = 0; i < count; ++i)
+            {
+                const auto current_point = getPoint(i);
```

Comment:
```suggestion
                const auto currentPoint = getPoint(i);
```
`clang-tidy` is failing because `current_point` is the wrong variable name style. Same goes for `previous_point`.